ffyaml/ffyaml.go

```diff
@@ -13,20 +13,84 @@ import (
 // Parser is a parser for YAML file format. Flags and their values are read
 // from the key/value pairs defined in the config file.
 func Parser(r io.Reader, set func(name, value string) error) error {
+	return New().Parse(r, set)
+}
+
+// ConfigFileParser is a parser for the YAML file format. Flags and their values
+// are read from the key/value pairs defined in the config file.
+// Nested nodes and keys are concatenated with a delimiter to derive the
+// relevant flag name.
+type ConfigFileParser struct {
+	delimiter string
+}
+
+// New constructs and configures a ConfigFileParser using the provided options.
+func New(opts ...Option) (c ConfigFileParser) {
+	c.delimiter = "."
+	for _, opt := range opts {
+		opt(&c)
+	}
+	return c
+}
+
+// Parse parses the provided io.Reader as a YAML file and uses the provided set function
+// to set flag names derived from the node names and their key/value pairs.
+func (c ConfigFileParser) Parse(r io.Reader, set func(name, value string) error) error {
```

Comment:
```suggestion
func (p *ConfigFileParser) Parse(r io.Reader, set func(name, value string) error) error {
```

(plus relevant changes to the comment)